pilot/pkg/xds/eds.go

```diff
@@ -165,9 +168,37 @@ func (s *DiscoveryServer) edsCacheUpdate(shard model.ShardKey, hostname string,
 	// For existing endpoints, we need to do full push if service accounts change.
 	if saUpdated {
 		log.Infof("Full push, service accounts changed, %v", hostname)
-		fullPush = true
+		return true, true
+	}
+	if created {
```

Comment:
For newly created shards, are we going to push the endpoints regardless of whether it is healthy or not? Seems like the logic below try to avoid pushing new endpoints which are not healthy yet, but here we are pushing regardless of that.

Reply:
Yes. That is correct. Because it is a full push, we want to send endpoints as well. The below is an optimization to reduce one additional endpoint push when a new pod comes up with with unready and changes to ready immediately.